plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -140,11 +140,14 @@ private static void sendPostRequest(String parametersString) {
       connection.setDoOutput(true);
       connection.setRequestMethod("POST");
       connection.setRequestProperty("Content-Length", Integer.toString(parametersString.length()));
+      connection.setReadTimeout(3000);  // milliseconds
 
       try (Writer writer = new OutputStreamWriter(connection.getOutputStream(), "UTF-8")) {
         writer.write(parametersString);
         writer.flush();
       }
+
+      connection.getResponseCode();  // Important: ensure this request posted.
```

Comment:
Might want to reference the issue URL in the code then. And I suggest language such as "Until we read, HTTP data may not be sent"